.github/workflows/connector-test-command.yml

```diff
@@ -69,64 +47,37 @@ jobs:
     name: Generate Connector Matrix
     runs-on: ubuntu-24.04
     steps:
-      - name: Get Connector List from Input
-        if: inputs.connector_name
-        run: |
-          # If input is newline-delimited, translate to comma-delimited
-          CONNECTORS=$(echo "${{ inputs.connector_name }}" | tr '\n' ',')
-          echo "Found connectors input:\n${CONNECTORS}"
-          echo "CONNECTORS=$CONNECTORS" >> $GITHUB_ENV
-
       - name: Checkout Current Branch
-        if: ${{ ! inputs.connector_name }}
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Get Connector List from Changes
-        if: ${{ ! inputs.connector_name }}
-        run: |
-          # Get modified connectors
-          CONNECTORS=$(./poe-tasks/get-modified-connectors.sh)
-          echo "Found connectors with changes:\n${CONNECTORS}"
-          # Convert newlines → commas (and drop any trailing comma)
-          CONNECTORS=$(printf '%s\n' "$CONNECTORS" \
-            | tr '\n' ',' \
-            | sed 's/,$//')
-          echo "Converted to comma-delimited list: ${CONNECTORS}"
-          echo CONNECTORS=${CONNECTORS} >> $GITHUB_ENV
-
-      - name: Convert to JSON
-        id: convert-matrix
+      - name: Generate Connector Matrix from Changes
+        id: generate-matrix
         run: |
-          # Check if CONNECTORS is empty
-          if [ -z "$CONNECTORS" ]; then
-            echo "No connectors modified. Setting matrix to empty string."
-            echo "matrix=" >> $GITHUB_OUTPUT
-            exit 0
-          fi
-          # quote each comma‑separated connector and rebuild the list in one step
-          json_elements="\"${CONNECTORS//,/\",\"}\""
-          json_output="{ \"connector\": [${json_elements}] }"
-          echo "matrix=$json_output" | tee -a "$GITHUB_OUTPUT"
+          # Get the list of modified connectors
+          echo "connectors_matrix=$(./poe-tasks/get-modified-connectors.sh --json)" | tee -a $GITHUB_OUTPUT
+          echo "jvm_connectors_matrix=$(./poe-tasks/get-modified-connectors.sh --json --java)" | tee -a $GITHUB_OUTPUT
+          echo "non_jvm_connectors_matrix=$(./poe-tasks/get-modified-connectors.sh --json --no-java)" | tee -a $GITHUB_OUTPUT
```

Comment:
This is greatly simplified now, since the shell script takes care of generating the JSON format needed to seed to the GitHub matrix.